modin/core/storage_formats/pandas/parsers.py

```diff
@@ -200,13 +204,30 @@ def generic_parse(fname, **kwargs):
             else:
                 for _ in range(header_size):
                     header += bio.readline()
-
+            if first_line is None:
+                # `first_line` can be b'' if end of file is reached
+                first_line = bio.readline()
             bio.seek(start)
-            to_read = header + bio.read(end - start)
+            to_read = bio.read(end - start)
+            header_with_to_read = header + to_read
         if "memory_map" in kwargs:
             kwargs = kwargs.copy()
             del kwargs["memory_map"]
-        pandas_df = callback(BytesIO(to_read), **kwargs)
+
+        try:
+            pandas_df = callback(BytesIO(header_with_to_read), **kwargs)
+        except pandas.errors.ParserError:
+            # looks like pandas determines the number of fields by the first line
+            # after the header, let's try like this and discard it after reading;
+            # as a test of the approach's operability, it's better to use it in some
+            # cases, not in all
+            if kwargs["quoting"] == csv.QUOTE_NONE and first_line:
+                pandas_df = callback(BytesIO(header + first_line + to_read), **kwargs)
```

Comment:
doesn't this tell us that `start` and `end` was computed incorrectly? is there a way to compute them properly at the beginning?

Reply:
Let's see how we split the test file:
```python
# first partition
a,b
0,"b0,b1"
1,"b2,b3"
# second partition
2,c
3,d
# third partition
4,"m,x"
5,"p,k"
```
It seems to me that the split is correct, just not good in terms of how pandas processes rows where there are not all fields.